controller/api/destination/watcher/endpoints_watcher.go

```diff
@@ -681,3 +914,38 @@ func diffAddresses(oldAddresses, newAddresses AddressSet) (add, remove AddressSe
 	}
 	return
 }
+
+func endpointSliceAccess(k8sClient kubernetes.Interface) bool {
+	gv := dv1beta1.SchemeGroupVersion.String()
+	res, err := k8sClient.Discovery().ServerResourcesForGroupVersion(gv)
+	if err != nil {
+		return false
+	}
+
+	if res.GroupVersion == gv {
+		for _, apiRes := range res.APIResources {
+			if apiRes.Kind == endpointSliceKind {
+				return true
+			}
+		}
+	}
+
+	return false
```

Comment:
I think we can move that somewhere else indeed. Not sure it is worth it before we actually need it outside of this file.